### ISSUE TITLE
[java] FinalFieldCouldBeStatic false-positive in Nested test class

### ISSUE THREAD

**Issue by Wolf2323** [pmd/pmd#5663]
**Affects PMD Version:** 7.11.0 (also tested with 7.12.0)

**Rule:** [FinalFieldCouldBeStatic](https://pmd.github.io/pmd/pmd_rules_java_design.html#finalfieldcouldbestatic)

**Description:**
Some time ago I reported this: https://github.com/pmd/pmd/issues/5018 and is actually solved now.
Since this is solved, I now face an other issue, that I guess is related to the fix of the previous issue.
Now I have fields, that are in a nested class, that is used by tests, but it gets now reported instead.

**Code Sample demonstrating the issue:**

```java
@ExtendWith(MockitoExtension.class)
class TimeProfilerTests {
	@Nested
	class operations_tests {
		private final long[] offset = {0L}; // <-- Here is the issue

		@Test
		void no_timespan_after_starting() {
			offset[0] += 1_000L;
			assertEquals(1000L, offset[0]);
		}
	}
```

**Expected outcome:**
This can not be static as is it used by multiple tests so there should be no violation. That's a false positive.

**Running PMD through:** *Maven*

**Comment by zbynek** [comment]
@Wolf2323 I think this is the same as #5858 -- the fact that the array initializer is in a nested class does not matter.

**Comment by oowekyala** [comment]
Thank you for noticing @zbynek
